File: packages/schema/tests/context.ts
/**
 * Shared context for the TypeScript package gates (suite split by concern).
 * Run the whole suite from packages/schema:
 *
 *     node --test "tests/*.test.ts"
 *
 * Zero third-party deps (node:test / node:assert / node:fs only; sha256 comes
 * from node:crypto inside gen/ids.ts).  The golden vectors in vectors.json
 * were generated ONCE from ids.py — gen/ids.ts reproducing them byte-for-byte
 * is the cross-language id-mint agreement gate.  VERDICT_CASES below is
 * duplicated VERBATIM from tests/context.py — the same table passing on both
 * implementations is the outline-policy agreement gate.
 */
import { readFileSync } from "node:fs";
import { fileURLToPath } from "node:url";
import { dirname, join } from "node:path";

export const PKG_ROOT = join(dirname(fileURLToPath(import.meta.url)), "..");
export const schemaObj = JSON.parse(readFileSync(join(PKG_ROOT, "schema.json"), "utf8"));
export const capabilityObj = JSON.parse(readFileSync(join(PKG_ROOT, "capability.json"), "utf8"));
const vectorsDoc = JSON.parse(readFileSync(join(PKG_ROOT, "vectors.json"), "utf8"));

export interface NodeVector {
  type: "node";
  input: { lang: string; kind: string; moduleName: string; rawName: string; file: string };
  expected: { canonicalName: string; path: string; preimage: string; id: string };
}
export interface EdgeVector {
  type: "edge";
  input: { kind: string; srcId: string; dstId: string };
  expected: { preimage: string; id: string };
}
export const vectors = vectorsDoc.vectors as (NodeVector | EdgeVector)[];

// Shared verdict cases — keep byte-identical with tests/context.py.
// [worstOf, expected worstToken, expected status, expected unrecognized]
export const VERDICT_CASES: [string[], string, string, string[]][] = [
  [["green"], "green", "green", []],
  [["lemma", "blue"], "blue", "blue", []],
  [["definition", "lemma"], "definition", "blue", []],
  [["red", "amber", "green"], "red", "red", []],
  [["none", "green"], "none", "green", []],
  [["amber", "blue"], "amber", "amber", []],
  [[], "none", "green", []],
  [["none/green"], "none/green", "unknown", ["none/green"]],
  [["green", "mystery-token"], "mystery-token", "unknown", ["mystery-token"]],
  [["red", "none/green"], "none/green", "unknown", ["none/green"]],
];

File: packages/schema/tests/context.py
"""Shared context for the schema package gates (suite split by concern).

Puts packages/schema and packages/schema/gen on sys.path, loads the canonical
JSON documents once, and carries the shared verdict-case table.  Run the whole
suite from packages/schema:

    python -m unittest discover -s tests

The verdict cases below are duplicated VERBATIM in tests/context.ts — the two
suites asserting the same table on both implementations is the cross-language
agreement gate for the outline policy (vectors.json plays the same role for
the id mint)."""
import json
import sys
from pathlib import Path

PKG_ROOT = Path(__file__).resolve().parents[1]
for p in (str(PKG_ROOT), str(PKG_ROOT / "gen")):
    if p not in sys.path:
        sys.path.insert(0, p)

SCHEMA_OBJ = json.loads((PKG_ROOT / "schema.json").read_text(encoding="utf-8"))
CAPABILITY_OBJ = json.loads((PKG_ROOT / "capability.json").read_text(encoding="utf-8"))
VECTORS = json.loads((PKG_ROOT / "vectors.json").read_text(encoding="utf-8"))

# Shared verdict cases — keep byte-identical with tests/context.ts.
# (worstOf, expected worstToken, expected status, expected unrecognized)
VERDICT_CASES = [
    (["green"], "green", "green", []),
    (["lemma", "blue"], "blue", "blue", []),
    (["definition", "lemma"], "definition", "blue", []),
    (["red", "amber", "green"], "red", "red", []),
    (["none", "green"], "none", "green", []),
    (["amber", "blue"], "amber", "amber", []),
    ([], "none", "green", []),
    (["none/green"], "none/green", "unknown", ["none/green"]),
    (["green", "mystery-token"], "mystery-token", "unknown", ["mystery-token"]),
    (["red", "none/green"], "none/green", "unknown", ["none/green"]),
]
